Appetize Integration (#903)
codemagic-ci-cd/codemagic-docs: content/integrations/appetize-integration.md

---
title: Appetize integration
description: How to integrate your workflows with Appetize using codemagic.yaml
weight: 11
---

[**Appetize**](https://appetize.io/) enables you to run native iOS and Android mobile apps directly in your browser. No downloads, plugins, or extra permissions needed. 

> ## Getting Started

Reviewer: ```suggestion
## Getting started
```

> Before getting started you need to get some tokens and keys.

Reviewer: ```suggestion
Before getting started you will need to generate an Appetize API token and a public key for your app.
```

> ### Get Appetize Token

Reviewer: ```suggestion
### Get Appetize token
```

> All users with admin or developer roles may request an API token after logging in at [account](https://appetize.io/account) page.

Reviewer: ```suggestion
All users with admin or developer roles may request an API token after logging in to your [account](https://appetize.io/account) page.
```

After getting your **API_TOKEN** you need to add it to your [environment variables](/variables/environment-variable-groups/#storing-sensitive-valuesfiles) in a group named **appetize** for example.

> ### Get Your App PublicKey

Reviewer: ```suggestion
### Get the public key for your app 
```

You have at least to upload your app once to [Appetize](https://appetize.io/upload) manually and then you can get the app *publicKey*.
in your *codemagic.yaml* file add the app key as a variable.

{{< highlight yaml "style=paraiso-dark">}}
    environment:
      groups:
        - appetize # <-- (Includes API_TOKEN)
      vars:

>         APP_PUBLIC_KEY: "YOUR_APP_PUBLIC_KEY"

Reviewer: Why set this locally in the YAML when it could be added to the `appetize` group that contains the API token?

{{< /highlight >}}

> ## Uploading Android App

Reviewer: ```suggestion
## Uploading an Android app
```

The following script will help you achieving the deployment process:

{{< highlight yaml "style=paraiso-dark">}}
- name: Publish APK to Appetize
  script: | 
    apkPath="/build/app/outputs/flutter-apk/app-release.apk"
    echo $(curl --location --request POST 'https://'$API_TOKEN'@api.appetize.io/v1/apps/'$APP_PUBLIC_KEY'' --form 'file=@"'$apkPath'"')
{{< /highlight >}}
Change the value of the `apkPath` to your actual apk path.

> ## Uploading iOS App

Reviewer: ```suggestion
## Uploading an iOS app
```

For iOS, you need to upload a `.zip` or `.tar.gz` file containing your compressed `.app` bundle.

> ### Xcode Build

Reviewer: ```suggestion
### Xcode build
```

You can run one of these commands to get your `.app`:
{{< highlight yaml "style=paraiso-dark">}}
- name: Build unsigned .app
  script: | 

>     xcodebuild -workspace "ios/Runner.xcworkspace" -scheme "$XCODE_SCHEME" -configuration "Debug" -sdk iphonesimulator -derivedDataPath ios/output
>     xcodebuild -project   "ios/Runner.xcodeproj"   -scheme "$XCODE_SCHEME" -configuration "Debug" -sdk iphonesimulator -derivedDataPath ios/output

Reviewer: ```suggestion
    xcodebuild -workspace "ios/Runner.xcworkspace" -scheme "$XCODE_SCHEME" -configuration "Debug" -sdk iphonesimulator -derivedDataPath ios/output
    xcodebuild -project   "ios/Runner.xcodeproj" -scheme "$XCODE_SCHEME" -configuration "Debug" -sdk iphonesimulator -derivedDataPath ios/output
```

Reviewer: Having both lines in the same script is confusing. I would comment one out, in case somebody blindly copies and pastes what you have put as the example.

{{< /highlight >}}

Then you need to zip the generated `.app` folder:

{{< highlight yaml "style=paraiso-dark">}}
- name: zip App.app
  script: | 
    cd ios/output/Build/Products/Debug-iphonesimulator
    zip -r ios_app.zip $XCODE_SCHEME.app
{{< /highlight >}}
Don't forget to set the value of your `XCODE_SCHEME` variable.

> ### Upload the .zip File

Reviewer: ```suggestion
### Upload the zip file
```
